@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -417,53 +419,42 @@ export class TopViewClipPlanesEvaluator extends ElevationBasedClipPlanesEvaluato
-            // sin(OCT) = sin(alpha) = r / d
-            // alpha = asin(r / d)
-            const alpha = Math.asin(r / d);
-            // If alpha it bigger than half fov angle, our visibility limit is set by tangent
-            // line, otherwise we need to find top (or right) plane intersection with sphere,
-            // which is definitely closer than the tangent point mentioned above.
-            const cam = camera as THREE.PerspectiveCamera;
-            // Take fov directly if it is vertical, otherwise we translate it using aspect ratio:
-
-            let halfFovAngle = THREE.MathUtils.degToRad(cam.fov / 2);
-            // If width > height, then we have to compute the horizontal FOV.
-            if (cam.aspect > 1) {
-                halfFovAngle = MapViewUtils.calculateHorizontalFovByVerticalFov(
-                    THREE.MathUtils.degToRad(cam.fov),
-                    cam.aspect
-                );
-            }
-
-            const maxR = r + this.maxElevation;
-            const farTangent = SphericalProj.getFarDistanceFromElevatedHorizon(cam, d, r, maxR);
-            farPlane =
-                halfFovAngle > alpha
-                    ? farTangent
-                    : this.getFovBasedFarPlane(cam, d, r, 2 * halfFovAngle, projection);
-        }
-        // Orthographic camera projection
-        else {
-            farPlane = this.getOrthoBasedFarPlane(d, r);
+
+        // This solution computes near and far plane for a set up where
+        // the camera is looking at the center of the scene.
+
+        // The far plane distance calculus requires finding the sphere tangent line that is
+        // co-linear with (goes thru) camera position, such tangent creates right angle
+        // with sphere diameter where it touches its surface (point T). Given that sphere is
+        // always at world origin and camera orbits around it we have (see
+        // #getTangentBasedFarPlane):
+        // angle(OTC) = 90
+        // sin(OCT) = sin(alpha) = r / d
+        // alpha = asin(r / d)
+        const alpha = Math.asin(r / d);
+        // If alpha it bigger than half fov angle, our visibility limit is set by tangent
+        // line, otherwise we need to find top (or right) plane intersection with sphere,
+        // which is definitely closer than the tangent point mentioned above.
+        // Take fov directly if it is vertical, otherwise we translate it using aspect ratio:
+
+        let halfFovAngle = THREE.MathUtils.degToRad(camera.fov / 2);
+        // If width > height, then we have to compute the horizontal FOV.
+        if (camera.aspect > 1) {
+            halfFovAngle = MapViewUtils.calculateHorizontalFovByVerticalFov(
+                THREE.MathUtils.degToRad(camera.fov),
+                camera.aspect
+            );
         }
 
+        const maxR = r + this.maxElevation;
+        const farTangent = SphericalProj.getFarDistanceFromElevatedHorizon(camera, d, r, maxR);
+        farPlane =
```

Comment:
I've removed the farPlane and nearPlane declarations at the top, couldn't make them const though const they are clamped below. We might be able to cleanup it up some more, but I'm not sure it's worth the time since this TopDownView evaluator is deprecated.